spec/rubocop/config_spec.rb

```diff
@@ -414,12 +388,87 @@
       end
 
       before do
-        allow(File).to receive(:file?).with('.ruby-version').and_return false
+        allow(File).to receive(:file?).with('.ruby-version')
       end
 
-      it 'falls back to TargetRubyVersion' do
+      it 'uses TargetRubyVersion' do
         expect(configuration.target_ruby_version).to eq ruby_version
       end
+
+      it 'does not read .ruby-version' do
+        configuration.target_ruby_version
+        expect(File).not_to have_received(:file?).with('.ruby-version')
+      end
+    end
+
+    context 'when TargetRubyVersion is unset' do
+      context 'when .ruby-version is present' do
+        before do
+          allow(File).to receive(:file?).with('.ruby-version').and_return true
+          allow(File)
+            .to receive(:read)
+            .with('.ruby-version')
+            .and_return ruby_version
+        end
+
+        context 'when .ruby-version contains an MRI version' do
+          let(:ruby_version) { '2.2.4' }
+          let(:ruby_version_to_f) { 2.2 }
+
+          it 'reads it to determine the target ruby version' do
+            expect(configuration.target_ruby_version).to eq ruby_version_to_f
+          end
+        end
+
+        context 'when the MRI version contains multiple digits' do
+          let(:ruby_version) { '10.11.0' }
+          let(:ruby_version_to_f) { 10.11 }
+
+          it 'reads it to determine the target ruby version' do
+            expect(configuration.target_ruby_version).to eq ruby_version_to_f
+          end
+        end
+
+        context 'when .ruby-version contains a version prefixed by "ruby-"' do
+          let(:ruby_version) { 'ruby-2.3.0' }
+          let(:ruby_version_to_f) { 2.3 }
+
+          it 'correctly determines the target ruby version' do
+            expect(configuration.target_ruby_version).to eq ruby_version_to_f
+          end
+        end
+
+        context 'when .ruby-version contains an JRuby version' do
```

Comment:
There’s (at least) maglev and mruby, too. I don’t know if we need test cases for all.